### PR TITLE
added .editorconfig.

### PR DESCRIPTION
**Pull request by phoenix24**
adding editor config, to help enforce consistent coding styles, quoting the [website](https://editorconfig.org) 

> EditorConfig helps maintain consistent coding styles for multiple developers working on the same project across various editors and IDEs. 

if the styles are agreed upon by maintainers, will follow up with another pull-request(s) enforcing these styles on the repository.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+indent_size = 2
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[*.md]
+trim_trailing_whitespace = false
+
+[*.java]
```
Comment (kingthorin) [inline .editorconfig]: This doesn't seem to align with the aosp format that is enforced in the builds etc.

**Comment by kingthorin** [comment]
With the gradle spotless tasks is this still needed?

**Comment by thc202** [comment]
It might be useful for the other file types, but if it's not enforced (automatically by the build) it's just a suggestion.

**Comment by kingthorin** [comment]
This is conflicting, doesn't conform to the spotless format which is used by the project, and seem abandon. Closing. (Please let us know if you decide to adapt it and want it re-opened).

**Pull request closed without merging**